src/Analysis/Ast/Impl/Analyzer/ModuleWalker.cs

```diff
@@ -246,118 +245,134 @@ public ModuleWalker(IServiceContainer services, IPythonModule module, PythonAst
                 var sourceType = sourceVar?.Value.GetPythonType();
 
                 // If stub says 'Any' but we have better type, keep the current type.
-                if (!IsStubBetterType(sourceType, stubType)) {
+                if (stubType.IsUnknown() || (stubType.DeclaringModule is TypingModule && stubType.Name == "Any")) {
```

Comment:
Previous code only worked when merging stub -> original. Now merge is original -> stub so most of the conditions no longer apply. I.e there is no `IsStubBetter` since now it is always better except in a couple of cases such as unknown or Any.

Reply:
`stubType.IsUnknown()` can be indeed removed since it is actually checked earlier.

Reply:
Ok then if you remove that check then it becomes clear enough. Do you mean that we are merging stub into original? Isn't that what we are still doing?

Reply:
No, we merge original into stub